### PR TITLE
locked user

### PR DESCRIPTION
**Pull request by itkg-nanne**
https://github.com/open-orchestra/open-orchestra-cms-bundle/pull/2342
https://github.com/open-orchestra/open-orchestra-user-bundle/pull/153

### CODE REVIEW

File: UserAdminBundle/Form/Type/UserType.php [hunk 1]
```diff
@@ -21,24 +23,28 @@ class UserType extends AbstractType
...
-        TranslatorInterface $translator
+        TranslatorInterface $translator,
+        AuthorizationCheckerInterface $authorizationChecker
     ) {
         $this->class = $class;
         $this->availableLanguages = $availableLanguages;
         $this->userProfileSubscriber = $userProfilSubscriber;
         $this->userGroupSubscriber = $userGroupSubscriber;
         $this->translator = $translator;
+        $this->allowedToSetPlatformAdmin = $authorizationChecker->isGranted(ContributionRoleInterface::PLATFORM_ADMIN);
```
Comment (alavieille) [inline UserAdminBundle/Form/Type/UserType.php]: not use authorization checker in construct

File: UserAdminBundle/Form/Type/UserType.php [hunk 2]
```diff
@@ -108,6 +114,14 @@ public function buildForm(FormBuilderInterface $builder, array $options)
                 ));
             $builder->addEventSubscriber($this->userProfileSubscriber);
             $builder->addEventSubscriber($this->userGroupSubscriber);
+            if ($this->allowedToSetPlatformAdmin || $this->allowedToSetDeveloper) {
+                $builder->add('accountLocked', 'checkbox', array(
```
Comment (alavieille) [inline UserAdminBundle/Form/Type/UserType.php]: why not use property enabled of fos user bundle ?

**Comment by alavieille** [comment]
where is check lock when user log in

File: UserAdminBundle/Form/Type/UserType.php (hunk 2 above)
Comment (itkg-nanne) [inline UserAdminBundle/Form/Type/UserType.php]: Is enabled specific to lock as there are : isCredentialsNonExpired, isAccountNonExpired ...?
In AdvancedUserInterface, you get a isEnabled method and a isAccountNonExpired method. So I'm not sure that using enabled is a good idea.

**Comment by itkg-nanne** [comment]
@alavieille : you didn't test ;)! the lock check is natively done by FOS. If you open 2 sessions in 2 browsers and lock one, next f5 or api call will result in logout.

**Comment by alavieille** [comment]
@itkg-nanne ok i check is natively by Symfony and not FOS in \Symfony\Component\Security\Core\User\UserChecker

**Comment by itkg-nanne** [comment]
upodate done @alavieille

File: UserAdminBundle/Form/Type/UserType.php [hunk 3]
```diff
@@ -47,6 +53,9 @@ public function __construct(
      */
     public function buildForm(FormBuilderInterface $builder, array $options)
     {
+        $allowedToSetPlatformAdmin = $this->authorizationChecker->isGranted(ContributionRoleInterface::PLATFORM_ADMIN);
```
Comment (alavieille) [inline UserAdminBundle/Form/Type/UserType.php]: move this check inside else

**Comment by itkg-nanne** [comment]
update done @alavieille

**Pull request merged**